### PR TITLE
Type Around: add information about how to type before/after a widget (to be read by screen readers)

### LINKED ISSUE #11936: Type Around feature is missing aria tags to be read properly by screen-readers
## 📝 Provide detailed reproduction steps (if any)

Based on support ticket np. 75352. Accessibility concern on SVG icons. The svg icons that are used in buttons should have an `alt` or `aria-label` or `role` tag(s) to be AA compliment.

### ✔️ Expected result

Names of buttons for Type Around feature should be discoverable by screen readers.

### ❌ Actual result

Names of buttons for Type Around feature are not discoverable by screen readers.

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

* Browser: Any
* OS: Any
* First affected CKEditor version: N/A
* Installed CKEditor plugins: widget

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by mmotyczynska**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (widget): Screen readers should now read the keyboard shortcuts to type around a widget. Closes #11936.

---

### Additional information

Known issues:

*   after plugin change:isEnabled, information in the label is updated not immediately but after the first selection change
*   changing readonly mode does not affect the label

### CODE REVIEW

**Comment by mlewand** [comment]
Based on what I know, though the solution have some edge cases - it solves most of the problems - so let's proceed with this proposal and extract the edge cases to a followup issue if it can't be fixed easily.

File: packages/ckeditor5-widget/src/utils.ts [hunk 1]
```diff
@@ -225,11 +227,12 @@ export function setHighlightHandling(
...
  *
  * @param {module:engine/view/element~Element} element
- * @param {String|Function} labelOrCreator
- * @param {module:engine/view/downcastwriter~DowncastWriter} writer
+ * @param {string|Function} labelOrCreator
  */
-export function setLabel( element: Element, labelOrCreator: string | ( () => string ), writer: DowncastWriter ): void {
-	writer.setCustomProperty( 'widgetLabel', labelOrCreator, element );
+export function setLabel( element: Element, labelOrCreator: string | ( () => string ) ): void {
+	const widgetLabel = element.getCustomProperty( 'widgetLabel' ) as Array<string | Function>;
```
Comment (niegowski) [inline packages/ckeditor5-widget/src/utils.ts:233]:
```suggestion
	const widgetLabel = element.getCustomProperty( 'widgetLabel' ) as Array<string | ( () => string )>;
```

File: packages/ckeditor5-widget/src/utils.ts [hunk 2]
```diff
@@ -239,13 +242,15 @@ export function setLabel( element: Element, labelOrCreator: string | ( () => str
  * @returns {String}
  */
 export function getLabel( element: Element ): string {
-	const labelCreator = element.getCustomProperty( 'widgetLabel' ) as string | ( () => string ) | undefined;
+	const widgetLabel = element.getCustomProperty( 'widgetLabel' ) as Array<string | Function>;
```
Comment (niegowski) [inline packages/ckeditor5-widget/src/utils.ts:245]:
```suggestion
	const widgetLabel = element.getCustomProperty( 'widgetLabel' ) as Array<string | ( () => string )>;
```

File: packages/ckeditor5-widget/src/utils.ts [hunk 3]
```diff
@@ -239,13 +242,15 @@ export function setLabel( element: Element, labelOrCreator: string | ( () => str
...
 
-	if ( !labelCreator ) {
-		return '';
-	}
-
-	return typeof labelCreator == 'function' ? labelCreator() : labelCreator;
+	return widgetLabel.reduce( ( prev: string, current: string | Function ) => {
```
Comment (niegowski) [inline packages/ckeditor5-widget/src/utils.ts:247]:
```suggestion
	return widgetLabel.reduce( ( prev: string, current: string | ( () => string ) ) => {
```

File: packages/ckeditor5-widget/src/widgettypearound/widgettypearound.ts [hunk 4]
```diff
@@ -256,9 +256,19 @@ export default class WidgetTypeAround extends Plugin {
...
 
+			if ( !viewElement ) {
+				return;
+			}
+
 			// Filter out non-widgets and inline widgets.
 			if ( isTypeAroundWidget( viewElement, data.item, schema ) ) {
 				injectUIIntoWidget( conversionApi.writer, buttonTitles, viewElement! );
+
+				const widgetLabel = viewElement.getCustomProperty( 'widgetLabel' ) as Array<string | Function>;
```
Comment (niegowski) [inline packages/ckeditor5-widget/src/widgettypearound/widgettypearound.ts:267]:
```suggestion
				const widgetLabel = viewElement.getCustomProperty( 'widgetLabel' ) as Array<string | ( () => string )>;
```

**Comment by niegowski** [comment]
What about this comment? https://github.com/ckeditor/ckeditor5/issues/11936#issuecomment-1249216002

**Comment by mmotyczynska** [comment]
> What about this comment? [#11936 (comment)](https://github.com/ckeditor/ckeditor5/issues/11936#issuecomment-1249216002)

I just added `aria-hidden` to these buttons, thanks for pointing that out :+1: 

Regarding the widget's `aria-describedby`, we added the description to the fake selection container instead because this is the one that is actually read by a screen reader when selecting a widget.

**Review by niegowski — APPROVED** [review]

**Comment by mlewand** [comment]
@niegowski are we waiting for something more or we're good to merge? 🙂

**Comment by niegowski** [comment]
> @niegowski are we waiting for something more or we're good to merge? 🙂

I don't know the "bigger picture" and if this is a part of sth bigger or a standalone PR that could be merged right away.

**Pull request merged**